### PR TITLE
feat: summarise console logs too

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

I've not written a ClickHouse alter migration, and wanted to do one while I had the opportunity to bother @yakkomajuri  about it (and I think @thmsobrmlr as new Yakko 🥇)

We collect console logs in session recordings but you can't filter to fid recordings that have them.

Worse, when we move to storing recordings in blob storage it would be impossible to filter for them.

## Changes

* adds new columns to the session replay summary table that will let us add counts of levels log, warn, and error from console logs collected with recordings
* alters the recordings-ingestion consumer to count those three levels of log

Screenshot 2023-06-08 at 21 56 20

TODO

- [x] return these counts from the API
- [x] filter by presence in the API
- [x] add UI to let someone filter and sort by them 

## How did you test this code?

running the migration locally and seeing I could still gather recordings

### CODE REVIEW

File: frontend/src/scenes/session-recordings/filters/SessionRecordingsFilters.tsx [hunk 1]
```diff
@@ -41,6 +48,51 @@ const filtersToLocalFilters = (filters: RecordingFilters): LocalRecordingFilters
...
+                label={'include console.warn'}
+            />
+            <LemonSwitch
+                size="small"
+                bordered={true}
+                checked={!!filters.console_logs?.includes('error')}
+                onChange={(checked) => updateChoice(checked, 'error')}
+                label={'include console.error'}
+            />
+        </div>
```
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/filters/SessionRecordingsFilters.tsx:92]: We _could_ use a multi select checkbox, like we do for the paths page. Feels a little more intuitive...
Screenshot 2023-06-09 at 08 21 02

File: posthog/models/session_replay_event/sql.py [hunk 2]
```diff
@@ -44,10 +47,23 @@
...
+# need to have the columns added, the SESSION_REPLAY_EVENTS_TABLE_BASE_SQL string
+# already add the columns
+# so, for e.g. test set up has them
+# Which means this is a no-op for new installations
+ALTER_SESSION_REPLAY_ADD_CONSOLE_COLUMNS = """
+    ALTER TABLE {table_name} on CLUSTER '{cluster}'
+        ADD COLUMN IF NOT EXISTS console_log_count SimpleAggregateFunction(sum, Int64),
+        ADD COLUMN IF NOT EXISTS console_warn_count SimpleAggregateFunction(sum, Int64),
+        ADD COLUMN IF NOT EXISTS console_error_count SimpleAggregateFunction(sum, Int64)
+"""
```
Comment (benjackwhite) [inline posthog/models/session_replay_event/sql.py:66]: Given that these commands are _only_ for migrations... Do we want to have them in a separate `migrations_sql.py` ? Just thinking that its a lot of clutter that is perhaps hard to parse for someone coming in and looking at this file

**Review by benjackwhite — CHANGES_REQUESTED** [review]

**Comment by pauldambra** [comment]
Screenshot 2023-06-09 at 09 48 27
Screenshot 2023-06-09 at 09 48 18
Screenshot 2023-06-09 at 09 48 13

**Review by benjackwhite — APPROVED** [review]
Approving but with some final "taste" comments that are up to you
